In [2]:
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
from numpy import int64
#from google.colab import files
  
#uploaded = files.upload()

vine_df = pd.read_csv("vine_table.csv")
vine_df.dtypes


vine_df.head()


review_id  star_rating  helpful_votes  total_votes vine  \
0   RSH1OZ87OYK92            2              2            3    N   
1  R1WFOQ3N9BO65I            5              0            0    N   
2   R3YOOS71KM5M9            5              0            0    Y   
3  R3R14UATT3OUFU            5              0            0    N   
4   RV2W9SGDNQA2C            5              0            0    Y   

  verified_purchase  
0                 N  
1                 Y  
2                 Y  
3                 Y  
4                 Y

In [4]:
vine_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 145431 entries, 0 to 145430
Data columns (total 6 columns):
 #   Column             Non-Null Count   Dtype 
---  ------             --------------   ----- 
 0   review_id          145431 non-null  object
 1   star_rating        145431 non-null  int64 
 2   helpful_votes      145431 non-null  int64 
 3   total_votes        145431 non-null  int64 
 4   vine               145431 non-null  object
 5   verified_purchase  145431 non-null  object
dtypes: int64(3), object(3)
memory usage: 6.7+ MB


## Filter by total votes


In [5]:
filtered_vine_df = vine_df[vine_df['total_votes'] >= 20]
filtered_vine_df.head()


review_id  star_rating  helpful_votes  total_votes vine  \
278    RV4UH0OWW97K4            5             19           22    N   
859    RG6GT6G9WNYT1            5             22           22    N   
924    RFBXE3Q3GYQYO            5             68           70    Y   
1436  R17V95WYC9ZXDD            1              0           20    Y   
1594  R235HEIYY15NZX            1              0           26    N   

     verified_purchase  
278                  N  
859                  N  
924                  Y  
1436                 Y  
1594                 Y

In [9]:
filtered_vine_df.count()

review_id            3342
star_rating          3342
helpful_votes        3342
total_votes          3342
vine                 3342
verified_purchase    3342
dtype: int64

## Find most helpful reviews

To retrieve all the rows where the number of helpful_votes divided by total_votes is equal to or greater than 50%


In [11]:
new_vine_df = filtered_vine_df.loc[filtered_vine_df.helpful_votes / filtered_vine_df.total_votes >= 0.5]
new_vine_df.head()

review_id  star_rating  helpful_votes  total_votes vine  \
278    RV4UH0OWW97K4            5             19           22    N   
859    RG6GT6G9WNYT1            5             22           22    N   
924    RFBXE3Q3GYQYO            5             68           70    Y   
1662  R34S64NPUZOS90            5             23           24    N   
1816  R35O3GV3HZUX8B            5             58           63    N   

     verified_purchase  
278                  N  
859                  N  
924                  Y  
1662                 N  
1816                 N

In [12]:
new_vine_df.count()

review_id            1685
star_rating          1685
helpful_votes        1685
total_votes          1685
vine                 1685
verified_purchase    1685
dtype: int64

## Reviews written as part of the vine program 


In [13]:
vine_reviews = new_vine_df[new_vine_df['vine'] == 'Y']
vine_reviews.head()


review_id  star_rating  helpful_votes  total_votes vine  \
924      RFBXE3Q3GYQYO            5             68           70    Y   
134300  R1I245TUZ85XLQ            2             22           27    Y   
134321   RSSQ9NOVPN8SN            5             25           33    Y   
134322  R1ELQB58Q36NQI            4             17           20    Y   
134352  R3262OBAU36BLD            1             28           38    Y   

       verified_purchase  
924                    Y  
134300                 Y  
134321                 N  
134322                 Y  
134352                 N

In [14]:
vine_reviews.count()

review_id            16
star_rating          16
helpful_votes        16
total_votes          16
vine                 16
verified_purchase    16
dtype: int64

## Reviews NOT written as part of the vine program 

In [15]:
not_vine_reviews = new_vine_df[new_vine_df['vine'] == 'N']
not_vine_reviews.head()


review_id  star_rating  helpful_votes  total_votes vine  \
278    RV4UH0OWW97K4            5             19           22    N   
859    RG6GT6G9WNYT1            5             22           22    N   
1662  R34S64NPUZOS90            5             23           24    N   
1816  R35O3GV3HZUX8B            5             58           63    N   
2189  R2AZAMZCEUOKQT            1             29           40    N   

     verified_purchase  
278                  N  
859                  N  
1662                 N  
1816                 N  
2189                 Y

In [16]:
not_vine_reviews.count()


review_id            1669
star_rating          1669
helpful_votes        1669
total_votes          1669
vine                 1669
verified_purchase    1669
dtype: int64

## 5-Star Review Analysis
Comparing the vine and non-vine program

In [10]:
count_vine = len(vine_reviews)
count_vine


16

In [11]:
count_nonvine = len(not_vine_reviews)
count_nonvine

1669

Out of all the reviews, there were 16 paid-vine reviews and 1,669 unpaid reviews.


5-star Reviews:

In [12]:
max_rating_vine = len(vine_reviews[vine_reviews['star_rating']==5])
max_rating_vine


6

In [13]:
max_rating_nonvine = len(not_vine_reviews[not_vine_reviews['star_rating']==5])
max_rating_nonvine

625

Out of those reviews, 6 of the vine reviews were 5 star and 625 were non-paid and 5 star


## 5-Star Review percentage by program

Percentage of 5-star Reviews:

In [14]:
max_rating_vine/count_vine


0.375

In [15]:
max_rating_nonvine/count_nonvine


0.3744757339724386

In [16]:
round(vine_reviews['star_rating'].mean(),2)


3.44

In [17]:
round(not_vine_reviews['star_rating'].mean(),2)


3.16